### PR TITLE
network: improve  the IsHost() function

### PR DESCRIPTION
**Pull request by obnoxxx**
This change is a follow-up to PR  https://github.com/rook/rook/pull/13693 .
It is also  meant to provide a continuation of the closed, stale PR https://github.com/rook/rook/pull/13878 .

Finally, it is  intended as a preparation for the introduction of a way to enforce host networking in the next round of updates to PR
 https://github.com/rook/rook/pull/13651 .

This change improves the documentation comment and the unit testing for the `IsHost()`  method of the `CephCluster.Network` spec.

The implementation of IsHost() is changed to validate the Network Spec and return error if the spec is invalid.

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [x] Documentation has been updated, if necessary.
- [x] Unit tests have been added, if necessary.
- [x] Integration tests have been added, if necessary.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -753,7 +753,7 @@ github.com/pmezard/go-difflib v1.0.1-0.20181226105442-5d4384ee4fb2 h1:Jamvg5psRI
 github.com/pmezard/go-difflib v1.0.1-0.20181226105442-5d4384ee4fb2/go.mod h1:iKH77koFhYxTK1pcRnkKkqfTogsbg7gZNVY4sRDYZ/4=
 github.com/portworx/dcos-secrets v0.0.0-20180616013705-8e8ec3f66611/go.mod h1:4hklRW/4DQpLqkcXcjtNprbH2tz/sJaNtqinfPWl/LA=
 github.com/portworx/kvdb v0.0.0-20200929023115-b312c7519467/go.mod h1:Q8YyrNDvPp3DVF96BDcQuaC7fAYUCuUX+l58S7OnD2M=
-github.com/portworx/sched-ops v0.20.4-openstorage-rc3 h1:tXnHsjZT2wZ2BCXf8avDoya7zGyCgLNUC8Upt+WEQrY=
+github.com/portworx/sched-ops v0.20.4-openstorage-rc3 h1:46EZ+vYCJ3qmQolvgDCrGuPz8Tf0vIds41RuF0dqVEw=
```
Comment (BlaineEXE) [inline go.sum:756]: Why is this changing? I don't see anything that appears related in the code changes.

This also seems to be the reason why CI is failing. Here's what I see when I look at the CI output:

```
...
verifying github.com/portworx/sched-ops@v0.20.4-openstorage-rc3: checksum mismatch
	downloaded: h1:tXnHsjZT2wZ2BCXf8avDoya7zGyCgLNUC8Upt+WEQrY=
	go.sum:     h1:46EZ+vYCJ3qmQolvgDCrGuPz8Tf0vIds41RuF0dqVEw=

SECURITY ERROR
This download does NOT match an earlier download recorded in go.sum.
The bits may have been replaced on the origin server, or an attacker may
have intercepted the download attempt.
```

File: pkg/apis/ceph.rook.io/v1/network.go [hunk 2]
```diff
@@ -56,7 +69,7 @@ func ValidateNetworkSpec(clusterNamespace string, spec NetworkSpec) error {
 	}
 
 	if !spec.AddressRanges.IsEmpty() {
-		if !spec.IsMultus() && !spec.IsHost() {
+		if !spec.IsMultus() && spec.Provider != NetworkProviderHost {
```
Comment (BlaineEXE) [inline pkg/apis/ceph.rook.io/v1/network.go:72]: I don't understand why this change is needed here. 

I think that this condition no longer works correctly if the user has `HostNetwork: true`, because the legacy behavior is no longer also supported.

File: pkg/apis/ceph.rook.io/v1/network_test.go [hunk 3]
```diff
@@ -72,10 +72,45 @@ func TestValidateNetworkSpec(t *testing.T) {
 	assert.NoError(t, err)
 }
 
-func TestNetworkCephIsHostLegacy(t *testing.T) {
-	net := NetworkSpec{HostNetwork: true}
+// test the NetworkSpec.IsHost method with different network providers
+// Also test it in  combination with the legacy
```
Comment (BlaineEXE) [inline pkg/apis/ceph.rook.io/v1/network_test.go:76]:
```suggestion
// Also test it in combination with the legacy
```

File: pkg/operator/ceph/cluster/mon/health.go [hunk 4]
```diff
@@ -670,7 +670,8 @@ func (c *Cluster) failoverMon(name string) error {
 		return errors.Wrap(err, "failed to place new mon on a node")
 	}
 
-	if c.spec.Network.IsHost() {
+	isHost, _ := c.spec.Network.IsHost()
```
Comment (BlaineEXE) [inline pkg/operator/ceph/cluster/mon/health.go:673]: We shouldn't make it a habit to ignore error return values.
Ditto for other similar changes as well.

File: pkg/apis/ceph.rook.io/v1/network_test.go [hunk 5]
```diff
@@ -72,10 +72,45 @@ func TestValidateNetworkSpec(t *testing.T) {
...
+// "HostNetwork  setting.
```
Comment (BlaineEXE) [inline pkg/apis/ceph.rook.io/v1/network_test.go:77]:
```suggestion
// "HostNetwork" setting.
```

File: pkg/apis/ceph.rook.io/v1/network_test.go [hunk 6]
```diff
@@ -72,10 +72,45 @@ func TestValidateNetworkSpec(t *testing.T) {
...
+	net.Provider = NetworkProviderHost
+	ret, err := net.IsHost()
+	assert.NoError(t, err)
+	assert.True(t, ret)
+
+	net.Provider = NetworkProviderDefault
+	net.HostNetwork = true
+	ret, err = net.IsHost()
+	assert.NoError(t, err)
+	assert.True(t, ret)
```
Comment (BlaineEXE) [inline pkg/apis/ceph.rook.io/v1/network_test.go:90]: Good unit test hygiene is to make sure each test sets up its own inputs so that future test modifications don't have unintended effects. It is hard to read and modify unit tests when test inputs and side-effects are propagated from between test cases.

Please fully specify `net = Network{}` before each test to ensure tests are independent from each other.

**Review by BlaineEXE — CHANGES_REQUESTED** [review]

File: go.sum (hunk 1 above)
Comment (obnoxxx) [reply go.sum:756]: @BlaineEXE  wrote:

> Why is this changing? I don't see anything that appears related in the code changes.

To be honest, I am not quite certain. I was puzzled myself when  I saw go.sum in the list of changed files. 🤷🏼

Here is my guess:

Earlier today, I wanted to build and unit-test locally and this was failing on go mod errors. So I tidied the dependencies and built/tested again, successfully. I assume that this has introduced the change to go.sum. 




> This also seems to be the reason why CI is failing.

Right, this seems plausible. Let me try and get rid of the go.sum change in the commit 




> Here's what I see when I look at the CI output:
> 
> ```
> ...
> verifying github.com/portworx/sched-ops@v0.20.4-openstorage-rc3: checksum mismatch
> 	downloaded: h1:tXnHsjZT2wZ2BCXf8avDoya7zGyCgLNUC8Upt+WEQrY=
> 	go.sum:     h1:46EZ+vYCJ3qmQolvgDCrGuPz8Tf0vIds41RuF0dqVEw=
> 
> SECURITY ERROR
> This download does NOT match an earlier download recorded in go.sum.
> The bits may have been replaced on the origin server, or an attacker may
> have intercepted the download attempt.
> ```

File: pkg/apis/ceph.rook.io/v1/network.go (hunk 2 above)
Comment (obnoxxx) [reply pkg/apis/ceph.rook.io/v1/network.go:72]: also note that the signature of `IsHost()`has changed to include an additional Error return

File: pkg/operator/ceph/cluster/mon/health.go (hunk 4 above)
Comment (obnoxxx) [reply pkg/operator/ceph/cluster/mon/health.go:673]: @BlaineEXE  wrote:

> We shouldn't make it a habit to ignore error return values. Ditto for other similar changes as well.

ok, good point. Agreed. I will address this, but it might make the code flow more complicated and less natural.

Comment (travisn) [reply pkg/operator/ceph/cluster/mon/health.go:673]: The complication is that `IsHost()` returns an error, but I don't think it should. The flow will be much simpler if `IsHost()` only returns a bool. There can be a single check in the reconcile for valid network settings, instead of every time `IsHost()` is called. This will also allow you to revert a lot of changes in this PR.

Comment (obnoxxx) [reply pkg/operator/ceph/cluster/mon/health.go:673]: @travisn  wrote: 
> The complication is that `IsHost()` returns an error, but I don't think it should.

The decision to have IsHost return error on invalid nrywork spec was a ewsult of the discussion between @BlaineEXE  and myself in the original PR #13878
https://github.com/rook/rook/pull/13878#discussion_r1513348424





>  The flow will be much simpler if `IsHost()` only returns a bool.
>
> There can be a single check in the reconcile for valid network settings, instead of every time `IsHost()` is called. 
Agreed.


> This will also allow you to revert a lot of changes in this PR.

This sounds like it is desirable to have as few changes as possible in one PR.
Of course, fewer changes tend to be simpler and therefore less likely to break things. So I do see the point. 😄

While it does seem kind of sad to me to revert a major piece of the work put into this so far, I will do it as I see the logic behind the reasoning.

Comment (obnoxxx) [reply pkg/operator/ceph/cluster/mon/health.go:673]: @travisn  starting to undo the changes that add the eroor return to IsHost(), I am wondering if it would make sense to still validate the network spec in IsHost() but just issue an informational log message and return false for host network instead of erroring out. wdyt?

File: go.sum (hunk 1 above)
Comment (obnoxxx) [reply go.sum:756]: Interestingly, when I locally revert the change to `go.sum` and run `make clean`and `go mod tidy`, I get the same error:

```console
verifying github.com/portworx/sched-ops@v0.20.4-openstorage-rc3: checksum mismatch
	downloaded: h1:46EZ+vYCJ3qmQolvgDCrGuPz8Tf0vIds41RuF0dqVEw=
	go.sum:     h1:tXnHsjZT2wZ2BCXf8avDoya7zGyCgLNUC8Upt+WEQrY=

SECURITY ERROR
This download does NOT match an earlier download recorded in go.sum.
The bits may have been replaced on the origin server, or an attacker may
have intercepted the download attempt.

For more information, see 'go help module-auth'.
```

so I am still unsure how to properly fix this.

**Comment by obnoxxx** [comment]
@BlaineEXE  wrote:

> One quick thing: The commit subject overflows the hard 70-character limit, and thus also overflows the recommended 50 char limit. Please reduce the commit subject title's length. IMO, "network: improve IsHost() function" is sufficient, as docs and unit tests are reasonable assumptions for any PR. Though you're free to title it as you like as long as it's strictly under 70 chars.

fixed, thanks!

File: go.sum (hunk 1 above)
Comment (obnoxxx) [reply go.sum]: I have undone the change to `go.sum` and `mod-check`passes in the ci now, but locally, `make mod.check`still fails.

**Comment by parth-gr** [comment]
I pulled the branch and ran `make build `

```
/home/rider/go/src/github.com/rook/rook/build/rbac
+ rm -rf /tmp/tmp.if5fFVidQV
+ popd
/home/rider/go/src/github.com/rook/rook
=== go vet
CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go vet  -tags '' github.com/rook/rook/cmd/... github.com/rook/rook/pkg/... github.com/rook/rook/tests/integration
diff pkg/apis/ceph.rook.io/v1/network.go.orig pkg/apis/ceph.rook.io/v1/network.go
--- pkg/apis/ceph.rook.io/v1/network.go.orig
+++ pkg/apis/ceph.rook.io/v1/network.go
@@ -46,8 +46,8 @@
 		err = errors.Wrapf(err, "Unable to determine  host network setting from Invalid  network spec. not setting host network.")
 	} else {
 
-	ret = (n.HostNetwork && n.Provider == NetworkProviderDefault) || n.Provider == NetworkProviderHost
-}
+		ret = (n.HostNetwork && n.Provider == NetworkProviderDefault) || n.Provider == NetworkProviderHost
+	}
 	return ret, err
 }
 
make[1]: *** [build/makelib/golang.mk:162: go.fmt] Error 1
make: *** [Makefile:124: build.common] Error 2
[root@fedora rook]# 

```

SO I think the error is coming from network API

File: pkg/apis/ceph.rook.io/v1/network.go [hunk 7]
```diff
@@ -32,10 +32,23 @@ func (n *NetworkSpec) IsMultus() bool {
...
-func (n *NetworkSpec) IsHost() bool {
-	return (n.HostNetwork && n.Provider == NetworkProviderDefault) || n.Provider == NetworkProviderHost
+// IsHost is intended to be used to determine if the rook operator should configure
+// managed pods to use host networking.
+// This behavior is enabled by configuring the cephCluster with the "host" network provider.
+// This method also maintains compatibility with the old HostNetwork setting
+// which is incompatible with other network providers: HostNetwork set to true
+// together with an empty or unset network provider has the same effect as
+// network.Provider set to "host"
+func (n *NetworkSpec) IsHost() (ret bool, err error) {
```
Comment (parth-gr) [inline pkg/apis/ceph.rook.io/v1/network.go:42]: I see there a lot of bugs with this implementation 
```
func (n *NetworkSpec) IsHost() (bool, error) {
	ret := false
	err := ValidateNetworkSpec("", *n)
	if err != nil {
		return ret, errors.Wrapf(err, "Unable to determine  host network setting from Invalid  network spec. not setting host network.")
	}
	return (n.HostNetwork && n.Provider == NetworkProviderDefault) || n.Provider == NetworkProviderHost, err
}
```

Fixing this make the ` make build` worked

**Review by parth-gr — CHANGES_REQUESTED** [review]

**Comment by obnoxxx** [comment]
@parth-gr  wrote:
> I pulled the branch and ran `make build `
> 
> ```
> /home/rider/go/src/github.com/rook/rook/build/rbac
> + rm -rf /tmp/tmp.if5fFVidQV
> + popd
> /home/rider/go/src/github.com/rook/rook
> === go vet
> CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go vet  -tags '' github.com/rook/rook/cmd/... github.com/rook/rook/pkg/... github.com/rook/rook/tests/integration
> diff pkg/apis/ceph.rook.io/v1/network.go.orig pkg/apis/ceph.rook.io/v1/network.go
> --- pkg/apis/ceph.rook.io/v1/network.go.orig
> +++ pkg/apis/ceph.rook.io/v1/network.go
> @@ -46,8 +46,8 @@
>  		err = errors.Wrapf(err, "Unable to determine  host network setting from Invalid  network spec. not setting host network.")
>  	} else {
>  
> -	ret = (n.HostNetwork && n.Provider == NetworkProviderDefault) || n.Provider == NetworkProviderHost
> -}
> +		ret = (n.HostNetwork && n.Provider == NetworkProviderDefault) || n.Provider == NetworkProviderHost
> +	}
>  	return ret, err
>  }
>  
> make[1]: *** [build/makelib/golang.mk:162: go.fmt] Error 1
> make: *** [Makefile:124: build.common] Error 2
> [root@fedora rook]# 
> ```
> 
> SO I think the error is coming from network API

right, and that is code from this PR.

I see the same error locally now, but the  confusing part is that this is a vet error
triggered from `make build` while `make vet`succeeds locally for me.

And the error message shows which code is wrong but not what aspect. ...

**Comment by obnoxxx** [comment]
> @parth-gr wrote:
> 
> > I pulled the branch and ran `make build `
> > ```
> > /home/rider/go/src/github.com/rook/rook/build/rbac
> > + rm -rf /tmp/tmp.if5fFVidQV
> > + popd
> > /home/rider/go/src/github.com/rook/rook
> > === go vet
> > CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go vet  -tags '' github.com/rook/rook/cmd/... github.com/rook/rook/pkg/... github.com/rook/rook/tests/integration
> > diff pkg/apis/ceph.rook.io/v1/network.go.orig pkg/apis/ceph.rook.io/v1/network.go
> > --- pkg/apis/ceph.rook.io/v1/network.go.orig
> > +++ pkg/apis/ceph.rook.io/v1/network.go
> > @@ -46,8 +46,8 @@
> >  		err = errors.Wrapf(err, "Unable to determine  host network setting from Invalid  network spec. not setting host network.")
> >  	} else {
> >  
> > -	ret = (n.HostNetwork && n.Provider == NetworkProviderDefault) || n.Provider == NetworkProviderHost
> > -}
> > +		ret = (n.HostNetwork && n.Provider == NetworkProviderDefault) || n.Provider == NetworkProviderHost
> > +	}
> >  	return ret, err
> >  }
> >  
> > make[1]: *** [build/makelib/golang.mk:162: go.fmt] Error 1
> > make: *** [Makefile:124: build.common] Error 2
> > [root@fedora rook]# 
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > SO I think the error is coming from network API
> 
> right, and that is code from this PR.
> 
> I see the same error locally now, but the confusing part is that

> this is a vet error triggered from `make build`

wrong it is in fact a gofmt error.



> And the error message shows which code is wrong but not what aspect.

manually running `go vet`yields more information:

```console:
$ go vet pkg/apis/ceph.rook.io/v1/network.go
# command-line-arguments
vet: pkg/apis/ceph.rook.io/v1/network.go:31:10: undefined: NetworkSpec
```

I am surprised that it complains only about line 31: There are many more occurrences like this.

File: pkg/operator/ceph/cluster/mon/health.go (hunk 4 above)
Comment (travisn) [reply pkg/operator/ceph/cluster/mon/health.go:673]: Since `IsHost()` is called from multiple locations, the log entry could easily be too verbose and repetitive. If there is an error condition to check for, I would recommend have a separate method that checks the error once during a reconcile, then `IsHost()` can assume there is no error condition and just return `bool`.

**Comment by BlaineEXE** [comment]
The go.sum file has been removed entirely in the commit. The build can't proceed without that

File: pkg/operator/ceph/cluster/mgr/spec.go [hunk 8]
```diff
@@ -106,7 +107,7 @@ func (c *Cluster) makeDeployment(mgrConfig *mgrConfig) (*apps.Deployment, error)
 	// Replace default unreachable node toleration
 	k8sutil.AddUnreachableNodeToleration(&podSpec.Spec)
 
-	if c.spec.Network.IsHost() {
+	if isHost := c.spec.Network.IsHost(); isHost {
```
Comment (travisn) [inline pkg/operator/ceph/cluster/mgr/spec.go:110]: no need for the var now, right?
```suggestion
	if c.spec.Network.IsHost() {
```

**Review by travisn — CHANGES_REQUESTED** [review]

**Comment by obnoxxx** [comment]
@BlaineEXE  wrote:

> The go.sum file has been removed entirely in the commit. The build can't proceed without that

interesting.

For me, locally, the build would not proceed or even succeed with the checked-in  `go.sum` file
It always produced a go module checksum mismatch error.
I had to remove the `go.sum` file  and run `go mod tidy`and a few more commands before I could build locally. That's why I committed the removal of the file.


Now I was able to fix the problem locally and build again. I committed the resulting diff and updated the PR.

File: go.sum (hunk 1 above)
Comment (travisn) [reply go.sum:756]: @obnoxxx Can you revert this file? It seems to be causing build issues.

File: pkg/operator/ceph/file/mds/spec.go [hunk 9]
```diff
@@ -63,7 +64,7 @@ func (c *Cluster) makeDeployment(mdsConfig *mdsConfig, fsNamespacedname types.Na
 			},
 			RestartPolicy:      v1.RestartPolicyAlways,
 			Volumes:            controller.DaemonVolumes(mdsConfig.DataPathMap, mdsConfig.ResourceName, c.clusterSpec.DataDirHostPath),
-			HostNetwork:        c.clusterSpec.Network.IsHost(),
+			HostNetwork:        useHost,
```
Comment (BlaineEXE) [inline pkg/operator/ceph/file/mds/spec.go:67]: If `IsHost()` is being reverted to only return a bool (even when there are configuration inconsistencies that Rook has to guess about), I don't think it makes sense to keep these changes that add `useHost := ...IsHost()`, as it's just code churn with no real changes. These should probably be reverted also.

At that point, the focus of the changes is better unit testing and docs, which is still valuable.

File: pkg/apis/ceph.rook.io/v1/network.go [hunk 10]
```diff
@@ -32,9 +32,14 @@ func (n *NetworkSpec) IsMultus() bool {
...
-// compatibility with the old HostNetwork field.
-func (n *NetworkSpec) IsHost() bool {
+// IsHost is intended to be used to determine if the rook operator should configure
+// managed pods to use host networking.
+// This behavior is enabled by configuring the cephCluster with the "host" network provider.
+// This method also maintains compatibility with the old HostNetwork setting
+// which is incompatible with other network providers: HostNetwork set to true
+// together with an empty or unset network provider has the same effect as
+// network.Provider set to "host"
+func (n *NetworkSpec) IsHost() (ret bool) {
```
Comment (travisn) [inline pkg/apis/ceph.rook.io/v1/network.go:42]: `ret` isn't referenced so it can be removed
```suggestion
func (n *NetworkSpec) IsHost() bool {
```

File: pkg/apis/ceph.rook.io/v1/zz_generated.deepcopy.go [hunk 11]
```diff
@@ -1186,6 +1186,11 @@ func (in *CephFilesystemSubVolumeGroupList) DeepCopyObject() runtime.Object {
 func (in *CephFilesystemSubVolumeGroupSpec) DeepCopyInto(out *CephFilesystemSubVolumeGroupSpec) {
 	*out = *in
 	in.Pinning.DeepCopyInto(&out.Pinning)
+	if in.Quota != nil {
```
Comment (travisn) [inline pkg/apis/ceph.rook.io/v1/zz_generated.deepcopy.go:1189]: These changes under `pkg/apis` actually are from master, not needed for your changes to `IsHost()`. I'd suggest just add these generated files in a separate commit in this PR. It seems our CI action is not working to catch that this generated code is out of date.

File: pkg/operator/ceph/cluster/mon/health.go [hunk 12]
```diff
@@ -670,7 +670,7 @@ func (c *Cluster) failoverMon(name string) error {
 		return errors.Wrap(err, "failed to place new mon on a node")
 	}
 
-	if c.spec.Network.IsHost() {
+	if  c.spec.Network.IsHost() {
```
Comment (travisn) [inline pkg/operator/ceph/cluster/mon/health.go:673]: revert this space, note that vscode would automatically remove these extra spaces for you, so you don't have to think about whitespace much in golang.

File: pkg/operator/ceph/cluster/mon/mon.go [hunk 13]
```diff
@@ -189,7 +189,7 @@ func (c *Cluster) Start(clusterInfo *cephclient.ClusterInfo, rookImage string, c
 	c.spec = spec
 
 	// fail if we were instructed to deploy more than one mon on the same machine with host networking
-	if c.spec.Network.IsHost() && c.spec.Mon.AllowMultiplePerNode && c.spec.Mon.Count > 1 {
+	if c.spec.Network.IsHost()  && c.spec.Mon.AllowMultiplePerNode && c.spec.Mon.Count > 1 {
```
Comment (travisn) [inline pkg/operator/ceph/cluster/mon/mon.go:192]: revert this space

File: pkg/operator/ceph/cluster/mon/mon.go [hunk 14]
```diff
@@ -923,7 +923,7 @@ func (c *Cluster) assignMons(mons []*monConfig) error {
 			// placement is not being made. otherwise, the node choice will map
 			// directly to a node selector on the monitor pod.
 			var schedule *controller.MonScheduleInfo
-			if c.spec.Network.IsHost() || c.monVolumeClaimTemplate(mon) == nil {
+			if c.spec.Network.IsHost()  || c.monVolumeClaimTemplate(mon) == nil {
```
Comment (travisn) [inline pkg/operator/ceph/cluster/mon/mon.go:926]: revert this space

File: pkg/operator/ceph/cluster/nodedaemon/pruner.go [hunk 15]
```diff
@@ -108,7 +108,7 @@ func (r *ReconcileNode) createOrUpdateCephCron(cephCluster cephv1.CephCluster, c
 				getCrashPruneContainer(cephCluster, *cephVersion),
 			},
 			RestartPolicy:      corev1.RestartPolicyNever,
-			HostNetwork:        cephCluster.Spec.Network.IsHost(),
+			HostNetwork:       cephCluster.Spec.Network.IsHost(),
```
Comment (travisn) [inline pkg/operator/ceph/cluster/nodedaemon/pruner.go:111]: revert this space

File: pkg/operator/ceph/file/mds/spec.go [hunk 16]
```diff
@@ -136,7 +135,7 @@ func (c *Cluster) makeMdsDaemonContainer(mdsConfig *mdsConfig, fsName string) v1
 		"--foreground",
 	)
 
-	if !c.clusterSpec.Network.IsHost() && !c.clusterSpec.Network.IsMultus() {
+	if !c.clusterSpec.Network.IsHost()  && !c.clusterSpec.Network.IsMultus() {
```
Comment (travisn) [inline pkg/operator/ceph/file/mds/spec.go:138]: revert this space

File: pkg/operator/ceph/object/spec_test.go [hunk 17]
```diff
@@ -707,7 +707,8 @@ func TestMakeRGWPodSpec(t *testing.T) {
 			if tt.isSet {
 				assert.Equal(t, *c.store.Spec.Gateway.HostNetwork, podTemplateSpec.Spec.HostNetwork)
 			} else {
-				assert.Equal(t, c.clusterSpec.Network.IsHost(), podTemplateSpec.Spec.HostNetwork)
+				isHost := c.clusterSpec.Network.IsHost()
+				assert.Equal(t, isHost, podTemplateSpec.Spec.HostNetwork)
```
Comment (travisn) [inline pkg/operator/ceph/object/spec_test.go:711]: you can revert this file as well

**Review by travisn — CHANGES_REQUESTED** [review]

File: pkg/operator/ceph/object/spec_test.go (hunk 17 above)
Comment (obnoxxx) [reply pkg/operator/ceph/object/spec_test.go]: done

File: pkg/operator/ceph/cluster/mon/health.go (hunk 12 above)
Comment (obnoxxx) [reply pkg/operator/ceph/cluster/mon/health.go]: done (manually 😉 )

File: pkg/operator/ceph/cluster/mon/mon.go (hunk 13 above)
Comment (obnoxxx) [reply pkg/operator/ceph/cluster/mon/mon.go]: done

**Review by obnoxxx — COMMENTED** [review]
addressed a few review requests and updated.

File: pkg/operator/ceph/controller/network.go [hunk 18]
```diff
@@ -49,7 +49,6 @@ func ApplyCephNetworkSettings(
 	clusterInfo *cephclient.ClusterInfo,
 ) error {
 	netSpec := clusterSpec.Network
-
```
Comment (travisn) [inline pkg/operator/ceph/controller/network.go:52]: could revert

File: pkg/operator/ceph/file/mds/spec.go [hunk 19]
```diff
@@ -101,7 +101,6 @@ func (c *Cluster) makeDeployment(mdsConfig *mdsConfig, fsNamespacedname types.Na
 			},
 		},
 	}
-
```
Comment (travisn) [inline pkg/operator/ceph/file/mds/spec.go:104]: could revert

**Review by travisn — APPROVED** [review]
LGTM, just a couple small whitespace changes that could be reverted if desired

File: pkg/operator/ceph/file/mds/spec.go (hunk 19 above)
Comment (obnoxxx) [reply pkg/operator/ceph/file/mds/spec.go]: done

File: pkg/operator/ceph/controller/network.go (hunk 18 above)
Comment (obnoxxx) [reply pkg/operator/ceph/controller/network.go]: done

**Comment by obnoxxx** [comment]
@travisn  wrote:

> LGTM, just a couple small whitespace changes that could be reverted if desired

Thanks! I addressed these

File: pkg/apis/ceph.rook.io/v1/network.go (hunk 7 above)
Comment (obnoxxx) [reply pkg/apis/ceph.rook.io/v1/network.go]: @parth-gr   this has been fixed.

File: go.sum (hunk 1 above)
Comment (obnoxxx) [reply go.sum]: done. no changes to `go.sum`in the PR anymore, even though I can't build locally without them.

**Review by BlaineEXE — APPROVED** [review]
